src/jit/emitxarch.cpp

```diff
@@ -13665,12 +13665,442 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+            }
+            break;
+
+        case INS_jo:
+        case INS_jno:
+        case INS_jb:
+        case INS_jae:
+        case INS_je:
+        case INS_jne:
+        case INS_jbe:
+        case INS_ja:
+        case INS_js:
+        case INS_jns:
+        case INS_jp:
+        case INS_jnp:
+        case INS_jl:
+        case INS_jge:
+        case INS_jle:
+        case INS_jg:
+            // Model a small chamce of a misprediction
+            result = PERFSCORE_INS_BRANCH_COND;
+            break;
+
+        case INS_i_jmp:
+            if (memFmt == IF_NONE)
+            {
+                result = PERFSCORE_INS_BRANCH_DIRECT;
+            }
+            else
+            {
+                result = PERFSCORE_INS_BRANCH_INDIRECT;
+            }
+            break;
+
+        case INS_call:
+            switch (insFmt)
+            {
+                case IF_LABEL:
+                    result = 3 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+
+                case IF_METHOD:
+                    result = 3 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+
+                case IF_METHPTR:
+                    result = 4 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+
+                case IF_SRD:
+                    result = 4 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+
+                case IF_ARD:
+                    result = 5 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+
+                default:
+                    assert(!"Unhandled insFmt for INS_call");
+                    result = PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+            }
+            break;
+
+        case INS_lock:
+            result = 13 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+            break;
+
+        case INS_xadd:
+        case INS_cmpxchg:
+            result = 5 * PERFSCORE_INS_THROUGHPUT_DEFAULT;
+            break;
+
+#ifdef _TARGET_X86_
+        case INS_fld:
+        case INS_fstp:
+            result = PERFSCORE_INS_THROUGHPUT_2X;
+            break;
+#endif // _TARGET_X86
+
+        case INS_movdqa:
+        case INS_movdqu:
+            switch (insFmt)
+            {
+                case IF_RWR_RRD:
+                    // ins   reg, reg
+                    result = PERFSCORE_INS_THROUGHPUT_4X;
+                    break;
+
+                case IF_RWR_ARD:
+                    // ins   reg, mem
+                    result = PERFSCORE_INS_THROUGHPUT_2X;
+                    break;
+
+                default:
+                    // all other insFmt's
+                    result = PERFSCORE_INS_THROUGHPUT_DEFAULT;
+                    break;
+            }
+            break;
+
+        case INS_xorps:
+            result = PERFSCORE_INS_THROUGHPUT_2X;
+            break;
+
+        default:
+            // all other ins
+            result = PERFSCORE_INS_THROUGHPUT_DEFAULT;
```

Comment:
This is going to negatively weight almost all code using hardware intrinsics and will not correctly weight some of the more expensive instructions.

Reply:
It is true that I didn't try to incorporate any reduced latencies for the hardware intrinsic instructions in this initial version.

Reply:
But this is typically going to be used to estimate code execution improvements for genearl optimization in the register allocator or with the CSE phase.  wherethe JIT has to made choices as to what sequence of instruction it is going to emit.

It is my understanding that when the Hardware Intrinsics are used the JIT must use these instructions.   If there are places in the JIT where it can chosebetween two different code sequences, wpne which uses SSE4 instructuions and another sequence which doesn't then 
we should definitely model those instructions, but I'm not surethere is any real benifit for modeling the exact latencies  for SSE4 instructions that the JIT is required to emit.

comparing Perf Scores when varying just the clrjit.dll is what this is aimed at.
comparing Perf Score when varying the input IL is **not** what this is aimed at.